src/Symfony/Component/Debug/ErrorHandler.php

```diff
@@ -160,6 +160,10 @@ public static function register($handler = null, $replace = true)
      */
     public function setDefaultLogger(LoggerInterface $logger, $levels = null, $replace = false)
     {
+        if (defined('FATAL_ERROR')) {
+            $this->loggers[FATAL_ERROR] = array(null, LogLevel::CRITICAL);
```

Comment:
This doesn't have this right side effect when the method is called several times for different levels. I'd suggest moving it to the constructor. I'd suggest also to add a new private property ($fatalErrors) that should be used instead of the current repeated calls to defined().